src/transformers/trainer.py

```diff
@@ -1224,6 +1258,28 @@ def train(
                     steps_trained_in_current_epoch -= 1
                     if steps_trained_progress_bar is not None:
                         steps_trained_progress_bar.update(1)
+                    if steps_trained_in_current_epoch == 0 and checkpoint_rng_state is not None:
+                        # We're finished skipping so set the RNG states to be exactly as they were at the
```

Comment:
```suggestion
                        # We have finished skipping so set the RNG states to be exactly as they were at the
```

Reply:
may I suggest that once polished we put all this RNG code into a separate function? This is currently making the training loop quite bloated. Thank you.